libsolidity/formal/CHC.cpp

```diff
@@ -707,6 +725,59 @@ void CHC::visitAddMulMod(FunctionCall const& _funCall)
 	SMTEncoder::visitAddMulMod(_funCall);
 }
 
+void CHC::visitDeployment(FunctionCall const& _funCall)
+{
+	if (!externalCallsIsTrustedMode())
+	{
+		SMTEncoder::endVisit(_funCall);
+		return;
+	}
+
+	auto funType = dynamic_cast<FunctionType const*>(_funCall.expression().annotation().type);
+	ContractDefinition const* contract =
+		&dynamic_cast<ContractType const&>(*funType->returnParameterTypes().front()).contractDefinition();
```

Comment:
Do you do this weird dereference juggling to have it throw on wrong types :-)?
Also do we want to be paranoid and assert that ```funType`` is not nullptr?
I'd just do sth like
```c++
ConstractDefinition const* contract = nullptr;
if (auto funType = dynamic_cast<FunctionType const*>(_funCall.expression().annotation().type))
   if (auto contractType = dynamic_cast<ContractType const*>(funType->returnParameterTypes().front()))
      contract = contractType->contractDefinition();
solAssert(contract, "");
```